internal/plugincommon/discovery/processor.go

```diff
@@ -53,19 +53,28 @@ func (cdp *ContractDiscoveryProcessor) Query(_ context.Context, _ dt.Outcome) (d
 func (cdp *ContractDiscoveryProcessor) Observation(
 	ctx context.Context, _ dt.Outcome, _ dt.Query,
 ) (dt.Observation, error) {
+	// all oracles should be able to read from the home chain, so we
+	// can fetch f chain reliably.
+	// TODO: should we error out here or just try to "observe everything we can"?
+	// i.e, similar to the commit plugin.
+	fChain, err := cdp.homechain.GetFChain()
+	if err != nil {
+		return dt.Observation{}, fmt.Errorf("unable to get fchain: %w", err)
+	}
+
 	contracts, err := (*cdp.reader).DiscoverContracts(ctx, cdp.dest)
 	if err != nil {
 		if errors.Is(err, reader.ErrContractReaderNotFound) {
-			// Not a dest reader, no observations will be made.
+			// Not a dest reader, only fChain observation will be made.
 			// Processor is not disabled because the outcome phase will bind observed contracts.
-			return dt.Observation{}, nil
+			return dt.Observation{
+				FChain: fChain,
+			}, nil
 		}
-		return dt.Observation{}, fmt.Errorf("unable to discover contracts: %w", err)
-	}
 
-	fChain, err := cdp.homechain.GetFChain()
-	if err != nil {
-		return dt.Observation{}, fmt.Errorf("unable to get fchain: %w", err)
+		// otherwise a legitimate error occurred when discovering.
+		// TODO: should we still return the f chain observation w/ a nil error?
+		return dt.Observation{}, fmt.Errorf("unable to discover contracts: %w", err)
```

Comment:
No. Let's not create a situation where we return an error but still need to handle the observation.